go/store/nbs/file_table_persister.go

```diff
@@ -208,3 +211,52 @@ func (ftp *fsTablePersister) ConjoinAll(ctx context.Context, sources chunkSource
 
 	return ftp.Open(ctx, name, plan.chunkCount, stats)
 }
+
+func (ftp *fsTablePersister) PruneTableFiles(ctx context.Context, contents manifestContents) error {
+	ss := contents.getSpecSet()
+
+	fileInfos, err := ioutil.ReadDir(ftp.dir)
+
+	if err != nil {
+		return err
+	}
+
+	err = ftp.fc.ShrinkCache()
+
+	if err != nil {
+		return err
+	}
+
+	for _, info := range fileInfos {
```

Comment:
If this encounters an `err`, should it accumulate them, and do as much work as possible?

Reply:
good idea